src/userobjects/ParticlesPerElementInitializer.C

```diff
@@ -0,0 +1,284 @@
+//* This file is part of FENIX: Fusion Energy Integrated Multiphys-X,
+//* A multiphysics application for modeling plasma facing components
+//* https://github.com/idaholab/fenix
+//*
+//* FENIX is powered by the MOOSE Framework
+//* https://www.mooseframework.inl.gov
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ParticlesPerElementInitializer.h"
+#include "ArbitraryQuadrature.h"
+#include "libmesh/enum_to_string.h"
+#include "MooseRandom.h"
+#include "Distribution.h"
+
+registerMooseObject("FenixApp", ParticlesPerElementInitializer);
+
+InputParameters
+ParticlesPerElementInitializer::validParams()
+{
+  auto params = InitializerBase::validParams();
+  params.addClassDescription(
+      "Particle initializer that uniformly distributes a specified number of particles per "
+      "element and calculates the corrisponding particle weight based on the requested number "
+      "density, particles per element and the elements \"volume\"");
+  params.addRequiredParam<unsigned int>(
+      "particles_per_element",
+      "The number of computational particles that should be placed in each element");
+  params.addRequiredParam<std::vector<DistributionName>>(
+      "velocity_distributions",
+      "The distribution names to be sampled when initializing the velocity of each particle");
+
+  params.addRequiredParam<Real>("number_density",
+                                "The number density of particles you want to represent");
+  return params;
+}
+
+ParticlesPerElementInitializer::ParticlesPerElementInitializer(const InputParameters & parameters)
+  : InitializerBase(parameters),
+    _number_density(getParam<Real>("number_density")),
+    _particles_per_element(getParam<unsigned int>("particles_per_element")),
+    _distribution_names(getParam<std::vector<DistributionName>>("velocity_distributions"))
+{
+  if (_particles_per_element <= 0)
+    paramError("particles_per_element", "The number of particles in each element must be >= 0");
+
+  if (_number_density == 0.0)
+    paramError("number_density", "The requested number density cannot be zero.");
```

Comment:
```suggestion
  if (_number_density <= 0.0)
    paramError("number_density", "The requested number density must be >= 0.");
```
Or could it really be negative?

Reply:
It could be depending on the user input. The number density should be a float and since there are no unsigned floats we need to perform a check for negative values.

Reply:
The number of particles per element on the other hand should never be zero so I will adjust that error message accordingly.